src/contents/en-gb/doc_dotnet.md

```diff
@@ -0,0 +1,265 @@
+
+We first need to generate an `AuthReq` instance which provides information about the application and, permissions requested by this application. These details will be displayed in the authenticator:
+```csharp
+var authReq = new AuthReq
+{
+    AppContainer = true,
+    App = new AppExchangeInfo 
+    { 
+        Id = "net.maidsafe.tutorial.dotnet", 
+        Scope = "", 
+        Name = "Hello SAFE Network", 
+        Vendor = "MaidSafe.net Ltd." 
+    },
+    Containers = new List<ContainerPermissions> 
+    { 
+        new ContainerPermissions 
+        { 
+            ContName = "_publicNames", Access = { Insert = true, Update = true, Delete = true} 
+        }
+    }
+};
+```
+Note that we are passing a list of `ContainerPermissions`, this allows us to define the various permissions provided to specific containers.
+
+When the Authenticator sends an authorisation response using the system URI mechanism, the system will match the URI scheme to our application, and it will launch our application passing the URI as an argument. Since we are running our application using the dotnet executable, we need to make sure the system registers the URI scheme with the dotnet executable file path. We are using system registry for this. Soon we will release .Net `system_uri` package to achieve this on all platforms.
+
+Once the `AuthReq` instance is initialised, we can use it to generate the authorisation request and send it to the Authenticator:
+```csharp
+var encodedReq = await Session.EncodeAuthReqAsync(authReq);
+var url = $"safe-auth://{encodedReq}";
+System.Diagnostics.Process.Start(url);
+```
+This code will generate the authorisation request and send it to the authenticator. Make sure you launch the Safe Browser and log in using the Authenticator ([install-safe-browser](#install-safe-browser)) before running the application.
+
+You would notice a pop-up on the Authenticator prompting authorisation request with the information of the application.
+
+At this point you can allow or deny the request, and the Authenticator will send the response back to the application. The demo application already has the code to send the response from new instance to already running console app instance using [IPC Pipes](https://docs.microsoft.com/en-us/dotnet/standard/io/pipe-operations).
+```csharp
+var decodeResult = await Session.DecodeIpcMessageAsync(encodedRequest);
+if (decodeResult.GetType() == typeof(AuthIpcMsg))
+{
+    var ipcMsg = decodeResult as AuthIpcMsg;
+    //Create session object
+    if (ipcMsg != null)
+    {
+        var session = await Session.AppRegisteredAsync(AppId, ipcMsg.AuthGranted);
+    }
+}
+else
+{
+    Console.WriteLine("Auth Request is not Granted");
+}
+```
+
+Once we get the response, we decode it and check whether the request was granted or denied. If the request was granted we will initialize a new session to communicate with SAFE Network.

```

Comment:
Once `a` ~we get the~ response `is received, it is ~we~ decode`d` ~it~ and check`ed` whether the request was granted or denied. If the request was granted ~we will initialize~ a new session `is initialised` to communicate with SAFE Network.